### ISSUE TITLE
relpFrame_t is not properly freed while handling protocol errors

### ISSUE THREAD

**Issue by vlnofka** [rsyslog/librelp#60]
Receiving relp frames are not correctly deallocated while handling protocol errors resulting in memory leak of dirty pages thus this issue is more severe issue than #59 especially over longer periods of time. 

Structure relpFrame_t isn't properly freed.

```
...
==4321==
==4321== 15,291,168 bytes in 159,283 blocks are definitely lost in loss record 31 of 31
==4321==    at 0x4C2DBC5: calloc (vg_replace_malloc.c:711)
==4321==    by 0x4E3F9FA: relpFrameConstruct (relpframe.c:51)
==4321==    by 0x4E3FBFB: relpFrameProcessOctetRcvd (relpframe.c:111)
==4321==    by 0x4E3E30A: relpSessRcvData (relpsess.c:253)
==4321==    by 0x4E3D8FB: doRecv (relp.c:552)
==4321==    by 0x4E3D8FB: handleSessIO (relp.c:710)
==4321==    by 0x4E3D8FB: engineEventLoopRun (relp.c:775)
==4321==    by 0x4E3DE37: relpEngineRun (relp.c:952)
==4321==    by 0x108F18: main (in /tmp/relp/librelp_receive)
==4321==
==4321== LEAK SUMMARY:
==4321==    definitely lost: 15,291,168 bytes in 159,283 blocks
==4321==    indirectly lost: 0 bytes in 0 blocks
==4321==      possibly lost: 192 bytes in 2 blocks
==4321==    still reachable: 96,385 bytes in 711 blocks
==4321==         suppressed: 0 bytes in 0 blocks
==4321==
```

Steps to reproduce:

- Use example code for receiving data or configure rsyslog for accepting relp connections.
```
# gcc receive.c -o librelp_receive -lrelp
# ./librelp_receive 20514
```
- Send relp messages with error. For example with this (and keep it running).
[relp_protocol_error.py.txt](https://github.com/rsyslog/librelp/files/834064/relp_protocol_error.py.txt)

- Process is going to increase it's allocated and used memory over time.

**Comment by rgerhards** [comment]
fixed via #77

**Comment by vlnofka** [comment]
Looks good. Thank you.
